app/common/helpers/collections.py

```diff
@@ -17,32 +19,61 @@ class CollectionHelper:
     """
 
     def __init__(self, collection: "Collection"):
-        self._collection = collection
+        """
+        Initialise the CollectionHelper; the `collection` instance passed in should have been retrieved from the DB
+        with the schema and related tables (eg section, form, question
+        :param collection:
+        """
+        self.collection = collection
+        self.schema = self.collection.collection_schema
 
     @classmethod
-    def load(cls, collection_id: UUID) -> "CollectionHelper":
-        return cls(get_collection(collection_id))
+    def load(cls, collection_id: uuid.UUID) -> "CollectionHelper":
+        return cls(get_collection(collection_id, with_full_schema=True))
 
     @property
     def grant(self) -> "Grant":
-        return self._collection.collection_schema.grant
-
-    @property
-    def schema(self) -> "CollectionSchema":
-        return self._collection.collection_schema
+        return self.schema.grant
 
     @property
     def sections(self) -> list["Section"]:
-        return self._collection.collection_schema.sections
+        return self.schema.sections
 
     @property
     def name(self) -> str:
-        return self._collection.collection_schema.name
+        return self.schema.name
 
     @property
     def status(self) -> str:
         return CollectionStatusEnum.NOT_STARTED
 
+    def get_section(self, section_id: uuid.UUID) -> "Section":
+        try:
+            return next(filter(lambda s: s.id == section_id, self.schema.sections))
```

Comment:
Just for my curiosity do you know if there's a performance or practical difference between `filter()` and lambda or list comprehension syntax? I've normally seen this written like

```python
next(s for s in self.schema.sections if s.id == section_id)
```

I'm fairly confident python is doing something clever behind the scenes between expanding the list and an iterable but I've not dug into it!